sunpy/data/data_manager/manager.py

```diff
@@ -0,0 +1,145 @@
+import pathlib
+import functools
+from contextlib import contextmanager
+
+from sunpy.util.util import hash_file
+
+
+class DataManager:
+    """
+    This class provides a remote data manager for managing remote files.
+
+    Parameters
+    ----------
+    cache: `sunpy.data.data_manager.cache.Cache`
+        Cache object to be used by `~sunpy.data.data_manager.manager.DataManager`.
+    """
+
+    def __init__(self, cache):
+        self._cache = cache
+
+        self._file_cache = {}
+
+        self._skip_hash_check = False
+        self._skip_file = {}  # Dict[str, str]
+
+    def require(self, name, urls, sha_hash):
+        """
+        Decorator for informing the data manager about the requirement of
+        a file by a function.
+
+        Parameters
+        ----------
+        name: `str`
+            The name to reference the file with.
+        urls: `list` or `str`
+            A list of urls to download the file from.
+        sha_hash: `str`
+            SHA-1 hash of file.
+        """
+        if isinstance(urls, str):
+            urls = [urls]
+        def decorator(func):
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                replace = self._skip_file.get(name, None)
+                err = KeyError("Hash does not match")
```

Comment:
I feel like it would be better to have the error say what the hash doesn't match. Maybe it would be better to just store the prefix string here?

Reply:
Can you explain what you mean by 'store the prefix string here'?

Reply:
I mean do something like:

```
err_message = "Hash does not match
```
here and then later down do:

```
raise KeyError(err_message + " because the world is on fire")
```